lib/player.js

```diff
@@ -3798,7 +3798,19 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
   addTextTrack(uri, language, kind, mime, codec, label) {
     // TODO: Add an actual error for this.
     if (this.loadMode_ == shaka.Player.LoadMode.SRC_EQUALS) {
-      shaka.log.error('Cannot add text when loaded with src=');
+      const trackElement = document.createElement('track');
+      trackElement.setAttribute('src', uri);
+      trackElement.setAttribute('label', label || '');
+      trackElement.setAttribute('kind', kind);
+      trackElement.setAttribute('srclang', language);
+      this.video_.appendChild(trackElement);
+      const textTracks = this.getTextTracks();
+      const track = textTracks.find((t) => t.language == language);
+      if (track) {
+        this.onTracksChanged_();
+        return track;
+      }
+      shaka.log.error('Cannot add this text when loaded with src=');
```

Comment:
I don't understand this condition.  What would it mean if track were null?  When/how would that occur?

Reply:
It is true that right now it does not work, but in the future not all the tracks that are added, will be reflected in src =

Natively, the browser only supports WebVTT, so for them to work in src = there are two alternatives:
1) Convert input subtitles to WebVTT (with new SRT PR it would be possible) and change the input url to a "local" URL with the converted subtitle.
2) Allow text streaming engine in src =

The first case will always work, but the subtitle support is more limited. The second case is to see how it works with HLS in Safari.

When the two PRs that I have in process merge, I will work on the first option.

Reply:
About the error... This should not happen, but there are browser implementations (in SmartTVs..) that may not support the Track element, so in that case an error is thrown. I will add a comment in the function.